lib/FusionInventory/Agent/Tools/Win32.pm

```diff
@@ -740,38 +769,38 @@ sub getInterfaces {
                     );
                 }
 
-                $interface->{GUID} = $object->{GUID}
-                    if $object->{GUID};
+                $interface->{GUID} = $object->{$netAdapterKey->{GUID}}
+                    if $object->{$netAdapterKey->{GUID}};
                 $interface->{DNSDomain} = $configuration->{DNSDomain}
                     if $configuration->{DNSDomain};
 
                 $interface->{SPEED}      = int($object->{Speed} / 1_000_000)
                     if $object->{Speed};
-                $interface->{VIRTUALDEV} = _isVirtual($object, $configuration);
+                $interface->{VIRTUALDEV} = _isVirtual($object, $configuration, $netAdapterKey);
 
                 push @interfaces, $interface;
             }
         } else {
             next unless $configuration->{MACADDR};
 
             my $interface = {
-                PNPDEVICEID => $object->{PNPDeviceID},
+                PNPDEVICEID => $object->{$netAdapterKey->{PNPDeviceID}},
                 PCIID       => $pciid,
                 MACADDR     => $configuration->{MACADDR},
-                DESCRIPTION => $configuration->{DESCRIPTION},
+                DESCRIPTION => exists $netAdapterKey->{DESCRIPTION} ? $object->{$netAdapterKey->{DESCRIPTION}} : $configuration->{DESCRIPTION},
                 STATUS      => $configuration->{STATUS},
                 MTU         => $configuration->{MTU},
                 dns         => $configuration->{dns}
             };
 
-            $interface->{GUID} = $object->{GUID}
-                if $object->{GUID};
+            $interface->{GUID} = $object->{$netAdapterKey->{GUID}}
+                if $object->{$netAdapterKey->{GUID}};
             $interface->{DNSDomain} = $configuration->{DNSDomain}
                 if $configuration->{DNSDomain};
 
             $interface->{SPEED}      = int($object->{Speed} / 1_000_000)
                 if $object->{Speed};
-            $interface->{VIRTUALDEV} = _isVirtual($object, $configuration);
+            $interface->{VIRTUALDEV} = _isVirtual($object, $configuration, $netAdapterKey);
```

Comment:
Here using a `FusionInventory::Agent::Tool::Win32::Network` would make even more sens and the `_isVirtual()` should even become a `->isVitual()` method of that class.